command/harness/setup.go

```diff
@@ -97,30 +94,43 @@ func HandleSetup(ctx context.Context, r *SetupVMRequest, s store.StageOwnerStore
 		return nil, fmt.Errorf("pool not defined")
 	}
 
+	cleanUpFn := func() {
+		errCleanUpStageOwner := s.Delete(context.Background(), stageRuntimeID)
+		if errCleanUpStageOwner != nil {
+			logr.WithError(errCleanUpStageOwner).Errorln("failed to delete failed stage owner")
+		}
+	}
+
+	if err := s.Create(ctx, &types.StageOwner{StageID: stageRuntimeID, PoolName: pool}); err != nil {
```

Comment:
we can update this line to first check if the entity is already present. If present, don't do anything. Same as we have done in destroy call. That will cover all the cases. No need to cleanup as it will happen in destroy anyways.

Reply:
The behaviour i've witness is if provisioning fails - the stage owner table isn't cleaned up. Yes the destroy call will clean it up now but that's not the behaviour the delegate is doing. Therefore this clean up necessary.